hatch_build.py

```diff
@@ -0,0 +1,83 @@
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+from hatchling.metadata.plugin.interface import MetadataHookInterface
+
+import tomlkit
+import pathlib
+
+
+class PinVersionMetadataHook(MetadataHookInterface):
+    """Update the dependency versions so they are pinned metadata of the wheel artefact produced by a wheel."""
+    def update(self, metadata):
+        change_count = 0
+        # Overlay the pinned dependencies, replacing dependencies if they already exist.
+        for pinned_dep in self.config["config"]["pinned_dependencies"]:
```

Comment:
Can you add a comment explaining that self.config is populated with the content of pyproject.toml which is tagged as tool.hatch.metadata.hooks.custom.config, again for anyone wondering later how this hangs together.

Reply:
Added extra info into the doctoring in the next commit.

Reply:
Looks great, suggested one more minor tweak.